File: day-1/part2.rb
#!/usr/bin/env ruby

tot = 0

File.read("input").chars.each_with_index do |char, i|
    tot +=
        case char
        when "(" then 1
        when ")" then -1
        else 0
        end
    if tot == -1
        puts(i + 1)
        exit
    end
end

File: day-1/part2.py
#!/usr/bin/env python

tot = 0

with open('input') as f:
    for i, char in enumerate(f.read()):
        if char == '(':
            tot += 1
        elif char == ')':
            tot -= 1
        if tot == -1:
            print i + 1
            exit()
